app/components/ak-pagination/index.ts

```diff
@@ -0,0 +1,75 @@
+import Component from '@glimmer/component';
+import { PaginationItemPerPageOptionProps } from '../ak-pagination-provider';
+import styles from './index.scss';
+
+export interface AkPaginationSignature {
+  Element: HTMLDivElement;
+  Args: {
+    disablePrev?: boolean;
+    disableNext?: boolean;
+    totalItems?: number;
+    startItemIdx?: number;
+    endItemIdx?: number;
+    itemPerPageOptions: PaginationItemPerPageOptionProps[];
+    selectedOption?: PaginationItemPerPageOptionProps;
+    onItemPerPageChange?: (
+      selectedItem: PaginationItemPerPageOptionProps
+    ) => void;
+    nextAction?: () => void;
+    prevAction?: () => void;
+    tableItemLabel?: string;
+    nextBtnLabel?: string;
+    prevBtnLabel?: string;
+    perPageTranslation?: string;
+    variant?: 'full' | 'compact';
+    paginationSelectOptionsVertPosition?: 'above' | 'below' | 'auto';
+  };
+}
+export default class AkPaginationComponent extends Component<AkPaginationSignature> {
+  COMMON_BUTTON_STYLES = {
+    height: '30px',
+    backgroundColor: 'white',
+    minWidth: 'unset',
+  };
+
+  noop() {}
+
+  get classes() {
+    return {
+      akPaginationSelect: styles['ak-pagination-select'],
+      akPrevButtonIconClass: styles['ak-pagination-button-prev-icon'],
+      akNextButtonIconClass: styles['ak-pagination-button-next-icon'],
+      compactButtonIconClass: styles['ak-pagination-compact-button-icon'],
+    };
+  }
+
+  get variant() {
+    return this.args.variant || 'full';
+  }
+
+  get isCompactPagination() {
+    return this.variant === 'compact';
+  }
+
+  get nextButtonStyles() {
+    return {
+      ...this.COMMON_BUTTON_STYLES,
+      paddingLeft: this.isCompactPagination ? '5px' : '12px',
+      paddingRight: this.isCompactPagination ? '5px' : '2px',
+    };
+  }
+
+  get prevButtonStyles() {
+    return {
+      ...this.COMMON_BUTTON_STYLES,
+      paddingLeft: this.isCompactPagination ? '5px' : '2px',
+      paddingRight: this.isCompactPagination ? '5px' : '12px',
+    };
+  }
+}
```

Comment:
this can be handled through css. `ak-pagination-variant-{{this.variant}}` and padding should be in em